utils/graph_sampler.jl

```diff
@@ -1,19 +1,30 @@
 using ABCDGraphGenerator
+using Random
 
-@info "Usage: julia graph_sampler.jl networkfile communityfile degreefile communitysizesfile μ isCL islocal"
-@info "Example: julia graph_sampler.jl network.dat community.dat degrees.dat community_sizes.dat 0.2 true true"
+@info "Usage: julia graph_sampler.jl networkfile communityfile degreefile communitysizesfile mu|xi fraction isCL islocal [seed]"
+@info "Example: julia graph_sampler.jl network.dat community.dat degrees.dat community_sizes.dat xi 0.2 true true 42"
 
 networkfile = ARGS[1]
 communityfile = ARGS[2]
 degreefile = ARGS[3]
 communitysizesfile = ARGS[4]
-μ = parse(Float64, ARGS[5])
-isCL = parse(Bool, ARGS[6])
-islocal = parse(Bool, ARGS[7])
+muxi = ARGS[5]
+fraction = parse(Float64, ARGS[6])
+isCL = parse(Bool, ARGS[7])
+islocal = parse(Bool, ARGS[8])
+
+muxi in ["mu","xi"] || throw(ArgumentError("only mu or xi names are allowed for"))
+μ = ξ = nothing
+if muxi == "mu"
+    μ = fraction
+else
+    ξ = fraction
+end
+length(ARGS) == 9 &&  Random.seed!(parse(Int, ARGS[9]))
 
 p = ABCDGraphGenerator.ABCDParams(parse.(Int, readlines(degreefile)),
                                   parse.(Int, readlines(communitysizesfile)),
-                                  μ, nothing, isCL, islocal)
+                                  μ, ξ, isCL, islocal)
```

Comment:
so again - we allow specifying ξ and passing islocal=true. I assume code throws an error later?

Reply:
Yes, it's handled in called code - again it can be handled sooner in above file if needed.